src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Tags/FluidTagHelper.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Fluid;
+using Fluid.Ast;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.DisplayManagement.Liquid.TagHelpers;
+
+namespace OrchardCore.DisplayManagement.Liquid.Tags
+{
+    public class FluidTagHelper
+    {
+        public static Dictionary<string, string> DefaultArgumentsMapping = new Dictionary<string, string>();
+
+        public static ValueTask<Completion> WriteArgumentsTagHelperAsync(List<FilterArgument> arguments, TextWriter writer, TextEncoder encoder, TemplateContext context)
+        {
+            return WriteToAsync(null, arguments, null, writer, encoder, context);
+        }
+
+        public static ValueTask<Completion> WriteArgumentsBlockHelperAsync(List<FilterArgument> arguments, IReadOnlyList<Statement> statements, TextWriter writer, TextEncoder encoder, TemplateContext context)
+        {
+            return WriteToAsync(null, arguments, statements, writer, encoder, context);
+        }
+
+        public static async ValueTask<Completion> WriteToAsync(string identifier, List<FilterArgument> arguments, IReadOnlyList<Statement> statements, TextWriter writer, TextEncoder encoder, TemplateContext context)
+        {
+            if (!context.AmbientValues.TryGetValue("Services", out var servicesValue))
+            {
+                throw new ArgumentException("Services missing while invoking 'helper'");
+            }
+
+            var services = servicesValue as IServiceProvider;
+
+            var viewContextAccessor = services.GetRequiredService<ViewContextAccessor>();
+            var viewContext = viewContextAccessor.ViewContext;
+
+            // If no identifier is set, use the first argument as the name of the tag helper
+            // e.g., {% helper "input", for: "Text", class: "form-control" %}
+
+            identifier = identifier ?? (await arguments[0].Expression.EvaluateAsync(context)).ToStringValue();
+
+            // These mapping will assign an argument name to the first element in the filter arguments,
+            // such that the tag helper can be matched based on the expected attribute names.
+            if (DefaultArgumentsMapping.TryGetValue(identifier, out var mapping))
+            {
+                arguments = new List<FilterArgument>(arguments);
+                arguments[0] = new FilterArgument(mapping, arguments[0].Expression);
```

Comment:
Should this be `.Add`?

Reply:
Ah it's initialized, with arguments, probably it has always > 0 items and cannot use the original list